[CI] JavaJodaTimeDuellingTests testIncompatiblePatterns failing

**Build Scans:**
- [elasticsearch-periodic #4546 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/g6likrcxps53m)
- [elasticsearch-periodic #4526 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/awywro64qbzfo)

**Reproduction Line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.common.joda.JavaJodaTimeDuellingTests.testIncompatiblePatterns" -Dtests.seed=AC6A0588A5B19E49 -Dtests.locale=ff-Adlm-SL -Dtests.timezone=Europe/Tiraspol -Druntime.java=23
```

**Applicable branches:**
7.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testIncompatiblePatterns'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.common.joda.JavaJodaTimeDuellingTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: "2019-01-01T01:01:01.001UTC"
     but: was "2019-01-01T01:01:01.001𞤑𞤖𞤊"
```

**Issue Reasons:**
- [7.17] 2 failures in test testIncompatiblePatterns (1.3% fail rate in 154 executions)
- [7.17] 2 failures in step openjdk23_java-matrix (40.0% fail rate in 5 executions)
- [7.17] 2 failures in pipeline elasticsearch-periodic (40.0% fail rate in 5 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/test/java/org/elasticsearch/common/joda/JavaJodaTimeDuellingTests.java
+++ b/server/src/test/java/org/elasticsearch/common/joda/JavaJodaTimeDuellingTests.java
@@ -43,8 +43,7 @@ public class JavaJodaTimeDuellingTests extends ESTestCase {
     public static void checkJvmProperties() {
         String localeProviders = System.getProperty("java.locale.providers", "");
         boolean runtimeJdk8 = JavaVersion.current().getVersion().get(0) == 8;
-        assert (runtimeJdk8 && localeProviders.equals("SPI,JRE"))
-            || (false == runtimeJdk8 && localeProviders.contains("SPI"))
+        assert (runtimeJdk8 && localeProviders.equals("SPI,JRE")) || (false == runtimeJdk8 && localeProviders.contains("SPI"))
             : "`-Djava.locale.providers` needs to be set";
         assumeFalse(
             "won't work in jdk8 because SPI mechanism is not looking at classpath - needs ISOCalendarDataProvider in jre's ext/libs",

--- a/server/src/test/java/org/elasticsearch/common/joda/JavaJodaTimeDuellingTests.java
+++ b/server/src/test/java/org/elasticsearch/common/joda/JavaJodaTimeDuellingTests.java
@@ -41,14 +41,16 @@ public class JavaJodaTimeDuellingTests extends ESTestCase {
 
     @BeforeClass
     public static void checkJvmProperties() {
+        String localeProviders = System.getProperty("java.locale.providers", "");
         boolean runtimeJdk8 = JavaVersion.current().getVersion().get(0) == 8;
-        assert (runtimeJdk8 && (System.getProperty("java.locale.providers", "").equals("SPI,JRE")))
-            || (false == runtimeJdk8 && (System.getProperty("java.locale.providers", "").contains("SPI")))
+        assert (runtimeJdk8 && localeProviders.equals("SPI,JRE"))
+            || (false == runtimeJdk8 && localeProviders.contains("SPI"))
             : "`-Djava.locale.providers` needs to be set";
         assumeFalse(
-            "won't work in jdk8 " + "because SPI mechanism is not looking at classpath - needs ISOCalendarDataProvider in jre's ext/libs",
+            "won't work in jdk8 because SPI mechanism is not looking at classpath - needs ISOCalendarDataProvider in jre's ext/libs",
             runtimeJdk8
         );
+        assumeFalse("CLDR messes up several datatime checks, ignore these", localeProviders.contains("CLDR"));
     }
 
     public void testIncorrectFormat() {

